### PR TITLE
Civ 2062 why do you disagree with the claim

### PR DESCRIPTION
**Pull request by laarmada**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-2062

### Change description ###

'Why do you disagree with the claim' screen

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/views/features/response/your-defence.njk [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+{% from "govuk/components/panel/macro.njk" import govukPanel %}
+
+{% block content %}
+  <div class="govuk-width-container">
+    <main class="govuk-main-wrapper" role="main">
+      {% set heading = 'Why do you disagree with the claim?' %}
+      {% set errors = form.getAllErrors() %}
+      {% set claimantFullName = claimantName %}
+      {#{% set theirReason = claim.claimData.reason %}#}
+      {% set theirReason = 'Reason your defence Reason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defence
```
Comment (commitBlob) [inline src/main/views/features/response/your-defence.njk:16]: was this left in intentionally?

File: src/main/views/features/response/your-defence.njk [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+{% from "../../macro/csrf.njk" import csrfProtection %}
+{% from "govuk/components/error-summary/macro.njk" import govukErrorSummary %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "../../macro/expandableSection.njk" import expandableSection %}
+{% from "govuk/components/panel/macro.njk" import govukPanel %}
+
+{% block content %}
+  <div class="govuk-width-container">
+    <main class="govuk-main-wrapper" role="main">
+      {% set heading = 'Why do you disagree with the claim?' %}
```
Comment (commitBlob) [inline src/main/views/features/response/your-defence.njk:12]: This line is missing translation

File: src/main/views/features/response/your-defence.njk [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+Reason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defence
+Reason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defence
+Reason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defence
+Reason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defence
+Reason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defenceReason your defence' %}
+      {% set labelContent =  '<p class="govuk-!-font-weight-bold">' + t("Briefly explain why you disagree with the claim") + '</p>' +
```
Comment (commitBlob) [inline src/main/views/features/response/your-defence.njk:24]: we've agreed to use keys for translations, could you refactor to be something like Tanya and myself did please.
https://github.com/hmcts/civil-citizen-ui/blob/CIV-2195_check_your_answers_pay_immediately/src/main/modules/i18n/locales/en.json
https://github.com/hmcts/civil-citizen-ui/blob/features/CIV-2488-reject-paid-less/src/main/modules/i18n/locales/en.json

File: src/test/unit/routes/features/response/yourDefenceController.test.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    nock(idamUrl)
+      .post('/o/token')
+      .reply(200, {id_token: citizenRoleToken});
+  });
+  describe('on Get', () => {
+    test('should return yourDefence page successfully', async () => {
+      app.locals.draftStoreClient = mockCivilClaim;
+      const claimantName = 'Mr. Jan Clark';
+      const inset = 'Your response will be sent to ' + claimantName + '.';
+      const header = 'Why do you disagree with the claim?';
```
Comment (commitBlob) [inline src/test/unit/routes/features/response/yourDefenceController.test.ts:32]: maybe you can remove header to the upper scope and have it declared once as it doesn't seem to change in the tests

**Review by commitBlob — COMMENTED** [review]
I can't see view unit tests?

**Review by LeonardoPalmeiro — CHANGES_REQUESTED** [review]

File: src/main/views/macro/expandableSection.njk [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{% macro expandableSection(label, content, limit = 400, id) %}
+  {% set longContent = content.length > limit %}
+
+  <div class="show-more-block">
+    <div class="govuk-!-font-weight-bold">{{ t(label) }}</div>
+    {% if longContent %}
+      <input id="{{ id }}" type="checkbox">
```
[the remark this answers is not in the document]
Comment (laarmada) [reply src/main/views/macro/expandableSection.njk:7]: that´s in the scope of the issue, show more and show less.

File: src/main/views/features/response/your-defence.njk [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                html: labelContent
+              },
+              errorMessage: {
+                text: t(form.errorFor('text'))
+              } if form.hasFieldError('text')
+            }) }}
+
+            <div class="govuk-inset-text">
+              <p class="govuk-body">
+                {{ t('Your response will be sent to {{ claimantFullName }}.', { claimantFullName: claimantFullName }) }}
```
[the remark this answers is not in the document]
Comment (laarmada) [reply src/main/views/features/response/your-defence.njk:60]: I just to check it and it´s mandatory else doesn´t work

File: src/main/views/features/response/your-defence.njk (hunk 1 above)
Comment (laarmada) [reply src/main/views/features/response/your-defence.njk:16]: done

File: src/test/unit/routes/features/response/yourDefenceController.test.ts (hunk 4 above)
Comment (laarmada) [reply src/test/unit/routes/features/response/yourDefenceController.test.ts:32]: it´s to maintain the same behaviour like in the view, the claimant name change depending of the type...

File: src/main/views/features/response/your-defence.njk (hunk 1 above)
Comment (laarmada) [reply src/main/views/features/response/your-defence.njk:16]: I have update retrieving information from correct data in redis.

**Review by GorilaNaranja — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by laperla11 — APPROVED** [review]

File: src/main/common/utils/languajeToogleUtils.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+const getLng = (lang: string | unknown): string => {
```
Comment (taomoh) [inline src/main/common/utils/languajeToogleUtils.ts:1]: typo in file name

**Review by taomoh — CHANGES_REQUESTED** [review]
typo in file name. It should be languageToggleUtils

File: src/main/common/utils/languajeToogleUtils.ts (hunk 7 above)
Comment (TatsyMaslova) [reply src/main/common/utils/languajeToogleUtils.ts:1]: And I think we already have this in languageToggleUitls:
const getLng = (lang: string | unknown): string => {
  return lang ? String(lang) : 'en';
};

export {
  getLng,
};

Comment (TatsyMaslova) [reply src/main/common/utils/languajeToogleUtils.ts:1]: Capture2

Comment (laarmada) [reply src/main/common/utils/languajeToogleUtils.ts:1]: two things, please don´t request change for an issue in PO review. On the other hand, this change come from master:
image

Comment (TatsyMaslova) [reply src/main/common/utils/languajeToogleUtils.ts:1]: Sorry Sergio but this change does not come from master. you misunderstood my comment. there is already a file with no spelling mistakes its called lanugageToggleUtils.ts Whereas in your pr you are introducing a new file with typo on the name that does exactly the same thing as languageToggleUtils.ts . If your pr goes to master we will have two files in utils that do the same thing and have the same export function.

Comment (laarmada) [reply src/main/common/utils/languajeToogleUtils.ts:1]: Hahaha I can see the error ok @taomoh I just to remove the wrong file...

**Review by taomoh — APPROVED** [review]

**Review by TatsyMaslova — APPROVED** [review]

**Pull request merged**